openedx/core/djangoapps/user_authn/views/logout.py

```diff
@@ -57,7 +57,11 @@ def dispatch(self, request, *args, **kwargs):
 
         logout(request)
 
-        response = super(LogoutView, self).dispatch(request, *args, **kwargs)
+        # If we don't need to deal with OIDC logouts, just redirect the user.
```

Comment:
This logic of looking at oauth_client_ids, I brought it back from Hawthorn. Without it, what happens is that studio will try to render a page that inherits from "main_django.html" and that template only exists in the LMS.

Now, I could also change the logic to make the `if` exclusively targeting studio and not all the cases where there are no oauth_clients. If that was the case, should I check for `SERVICE_VARIANT == 'cms'` or is there a preferred way of doing this?

Reply:
The rendered page is a harmless, but ugly looking bug in the master version. You can see what I mean if you go to https://studio.edx.org/logout (even without a valid session)

Reply:
Also this part here is what is breaking the tests. I'll fix the tests once I have a clear indication about what I should do in this case

Reply:
I believe @pwnage101 and @doctoryes had reasons to remove this code as part of supporting SSO via DOT (rather than DOP).

@felipemontoya Can you put this logic behind the `DISABLE_STUDIO_SSO_OVER_LMS` feature toggle? When Studio redirects to the LMS for both login/logout, then Studio's own logout page should never be rendered and the ugly page you point out should not appear.  Eventually, we will want to remove the old studio login/logout code entirely, including this now-ugly logout page.

Reply:
This makes sense to me. I just made the necessary changes.
I also rebased on top of master to have a more up-to-date PR